include/experimental/__p0009_bits/extents.hpp

```diff
@@ -306,13 +329,72 @@ class extents
       std::conditional_t<N==rank_dynamic(),
         detail::__construct_psa_from_dynamic_exts_array_tag_t<0>,
         detail::__construct_psa_from_all_exts_array_tag_t>(),
-        std::array<SizeType,N>{exts}
+      std::array<SizeType,N>{exts}
 #if defined(_MDSPAN_USE_ATTRIBUTE_NO_UNIQUE_ADDRESS)
       }
 #else
       }})
 #endif
-  { }
+  {
+    /* TODO: precondition check
+     * If N != rank_dynamic() is true, exts[r] equals Er for each r for which Er is a static extent, and
+     * either
+     *   - N is zero, or
+     *   - exts[r] is nonnegative and is a representable value of type size_type for all rank index r
+     */
+  }
+
+#if MDSPAN_HAS_CXX_20
+  // TODO: check whether this works with newest NVCC, doesn't with 11.4
+#ifdef __NVCC__
+  // NVCC seems to pick up rank_dynamic from the wrong extents type???
+  // NVCC chokes on the fold thingy here so wrote the workaround
+  MDSPAN_TEMPLATE_REQUIRES(
+    class SizeType, size_t N,
+    /* requires */ (
+      _MDSPAN_TRAIT(is_convertible, SizeType, size_type) &&
+      _MDSPAN_TRAIT(is_nothrow_constructible, size_type, SizeType) &&
+      ((N == detail::_count_dynamic_extents<Extents...>::val) ||
+       (N == sizeof...(Extents)))
+    )
+  )
+#else
+    MDSPAN_TEMPLATE_REQUIRES(
+        class SizeType, size_t N,
+        /* requires */ (
+          _MDSPAN_TRAIT(is_convertible, SizeType, size_type) &&
+          _MDSPAN_TRAIT(is_nothrow_constructible, size_type, SizeType) &&
+          (N == rank() || N == rank_dynamic())
+    )
+  )
+#endif
+  MDSPAN_CONDITIONAL_EXPLICIT(N != rank_dynamic())
+  MDSPAN_INLINE_FUNCTION
+  constexpr
+  extents(std::span<SizeType, N> const& exts) noexcept
```

Comment:
```suggestion
  extents(std::span<SizeType, N> exts) noexcept
```
The current version of the proposal in the main branch passes `span` by value, not const reference.

Reply:
yeah will change